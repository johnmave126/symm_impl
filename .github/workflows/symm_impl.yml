name: Test

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  test:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v1

      - name: Install rust toolchain
        uses: actions-rs/toolchain@v1
        with:
          toolchain: stable
          override: true
          default: true

      - name: Restore Cache
        uses: actions/cache@v2
        with:
          path: |
            ~/.cargo/registry
            ~/.cargo/git
            target
          key: test-cargo-${{ hashFiles('**/Cargo.toml') }}

      - name: Quick check
        run: cargo check

      - name: Test
        run: cargo test