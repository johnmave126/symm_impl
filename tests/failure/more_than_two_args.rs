use symm_impl::symmetric;

trait T<Other> {
    fn op(&self, other: &Other, more: i32) -> i32;
}

struct A {
    a: i32,
}

struct B {
    b: i32,
}

#[symmetric]
impl T<B> for A {
    fn op(&self, other: &B, more: i32) -> i32 {
        self.a + other.b
    }
}

fn main() {}
