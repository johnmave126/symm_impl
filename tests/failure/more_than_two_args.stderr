error: expected 2 arguments
  --> $DIR/more_than_two_args.rs:17:11
   |
17 |     fn op(&self, other: &B, more: i32) -> i32 {
   |           ^
